modules/ximgproc/include/opencv2/ximgproc/ridgefilter.hpp

```diff
@@ -0,0 +1,45 @@
+// This file is part of OpenCV project.
+// It is subject to the license terms in the LICENSE file found in the top-level directory
+// of this distribution and at http://opencv.org/license.html.
+/*
+Ridge Detection Filter.
+OpenCV port by : Kushal Vyas (@kushalvyas), Venkatesh Vijaykumar(@venkateshvijaykumar)
+*/
+
+#ifndef __OPENCV_XIMGPROC_RIDGEFILTER_HPP__
+#define __OPENCV_XIMGPROC_RIDGEFILTER_HPP__
+
+#include <opencv2/core.hpp>
+
+namespace cv{
+namespace ximgproc{
+    //! @addtogroup ximgproc_filters
+    //! @{
+    class CV_EXPORTS_W RidgeDetectionFilter : public Algorithm{
+    /**
+    @brief  Applies Ridge Detection Filter to an input image. Implements Ridge detection similar to the one in [Mathematica](http://reference.wolfram.com/language/ref/RidgeFilter.html) using the eigen values from the Hessian Matrix of the input image using Sobel Derivatives.Additional refinement can be done using Skeletonization and Binarization.
```

Comment:
The brief description should be located right before declaration of `RidgeDetectionFilter`.